packages/babel-plugin-transform-spread/src/index.ts

```diff
@@ -33,15 +44,41 @@ export default declare((api, options) => {
     return false;
   }
 
-  function push(_props, nodes) {
+  function push(
+    _props: Array<
+      | t.Expression
+      | t.SpreadElement
+      | t.JSXNamespacedName
+      | t.ArgumentPlaceholder
+    >,
+    nodes: Array<t.Expression>,
+  ): Array<
+    t.Expression | t.SpreadElement | t.JSXNamespacedName | t.ArgumentPlaceholder
+  > {
     if (!_props.length) return _props;
-    nodes.push(t.arrayExpression(_props));
+    nodes.push(
+      t.arrayExpression(_props as Array<t.Expression | t.SpreadElement>),
```

Comment:
If this only works when `_props` is `Array<t.Expression | t.SpreadElement>`, it should be reflected in the function signature.